### PR TITLE
feat: add javdb session config and cookieJar to fakeGot

### LINKED ISSUE #14423: javdb empty route
### 路由地址

```routes
/javdb/actors/:id/:filter?
```

### 完整路由地址

```fullroutes
/javdb/actors/Xa1G/d
```

### 相关文档

https://docs.rsshub.app/routes/multimedia#javdb-zhu-ye

### 预期是什么？

返回正确结果

### 实际发生了什么？

实际显示this route is empty
image

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

docker  diygod/rsshub:chromium-bundled:latest

### 额外信息

```shell
none
```

### 这不是重复的 issue

- [X] 我已经搜索了 [现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by hqwuzhaoyi**
## Involved Issue / 该 PR 相关 Issue

Close #14423

## Example for the Proposed Route(s) / 路由地址示例

```routes
/javdb/actors/Xa1G/d
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

添加了环境变量JAVDB_SESSION，用于获取用于有些页面需要登录,会获取数据失败

该SESSION因为javdb的登录政策，可能随着用户登出账户失效，建议保留登录状态 

删除了 lib/routes/javdb/index.ts 下的 '/:category?/:sort?/:filter?'，会导致其他子模块路由无法进入

### CODE REVIEW

File: lib/routes/javdb/series.ts [hunk 1]
```diff
@@ -46,5 +46,7 @@ async function handler(ctx) {
 
     const title = `JavDB${filters[filter] === '' ? '' : ` - ${filter[filter]}`} `;
 
-    return await utils.ProcessItems(ctx, currentUrl, title);
+    const data = await utils.ProcessItems(ctx, currentUrl, title);
+
+    return data;
```
Comment (TonyRL) [inline lib/routes/javdb/series.ts:51]: Unnecessary changes

File: lib/routes/javdb/tags.ts [hunk 2]
```diff
@@ -43,5 +43,7 @@ async function handler(ctx) {
 
     const title = `JavDB${query === '' ? '' : ` - ${query}`} `;
 
-    return await utils.ProcessItems(ctx, currentUrl, title);
+    const data = await utils.ProcessItems(ctx, currentUrl, title);
+
+    return data;
```
Comment (TonyRL) [inline lib/routes/javdb/tags.ts:48]: Unnecessary changes

**Review by TonyRL — COMMENTED** [review]
cookieJar is no longer available with #14922

**Comment by hqwuzhaoyi** [comment]
> cookieJar is no longer available with #14922

In the fakeGot inside got, a parameter cookieJar has been added to allow for adding cookies. Please take a look at this implementation to see if it is feasible.

File: lib/routes/javdb/actors.ts [hunk 3]
```diff
@@ -46,6 +46,5 @@ async function handler(ctx) {
     };
 
     const title = `JavDB${filters[filter] === '' ? '' : ` - ${filters[filter]}`} `;
-
```
Comment (TonyRL) [inline lib/routes/javdb/actors.ts:49]: Please also update
https://github.com/DIYgod/RSSHub/blob/5e142390629cd8c0b1fc4a49fd6cc59b9acedce5/lib/routes/javdb/actors.ts#L10
to
```ts
    features: {
        requireConfig: [{
            name: 'JAVDB_SESSION',
            description: 'Description here',
            optional: true,
        }],
```

Comment (hqwuzhaoyi) [reply lib/routes/javdb/actors.ts:49]: Thank you for your guidance, I have made the changes, please review again.

**Comment by SecondaryH** [comment]
How can I add the token to a docker compose file? I currently got 

`JAVDB_SESSION: inserttokenhere`

but it isnt working

**Comment by hqwuzhaoyi** [comment]
> How can I add the token to a docker compose file? I currently got
> 
> `JAVDB_SESSION: inserttokenhere`
> 
> but it isnt working

You can directly update the docker compose like below and pull the latest image. I just verified it, and it should work.

```
version: '3.9'

services:
    rsshub:
        # two ways to enable puppeteer:
        # * comment out marked lines, then use this image instead: diygod/rsshub:chromium-bundled
        # * (consumes more disk space and memory) leave everything unchanged
        image: diygod/rsshub
        restart: always
        ports:
            - '1200:1200'
        environment:
            NODE_ENV: production
            CACHE_TYPE: redis
            REDIS_URL: 'redis://redis:6379/'
            PUPPETEER_WS_ENDPOINT: 'ws://browserless:3000'  # marked
            JAVDB_SESSION: "rb%2BTUQwgHBCgCinOZ4iMwQGbVgFPD%2BWDBHCiaQVsQc8XsnatbBRXa7M2yPeyR1uAwICtmZV%2BKKbkC%2F9souH%2FriNxF90USd0VuLPDNZUw%2xxxxxxxxxxxxx4nKC7CZS9VX%2BWy%2BUp8kphSbgjXiQCMoKRvvWISQ7CWrM%2BnJcXARZyNMS7VIzHPG6kTD73aP4VMcOUs%2FL6uu%2F45LNGvg9JbJAuOOn43WNmZSfzcEXjKu0ocj3JTa1kJCucIXOd40Y95HDYYz56ceEhFXXCLxw47nib8iQ0jgcYQVvCDfTospjgrSg5z5ouf5j00IHZjslMoP%2F0ZFupSv6OmZmYY3JZopgaz8%3D--hm%2FMiUKczzf2tQsL--JTR3IA8lGh%2BVud%2F3I3UOQg%3D%3D"
        depends_on:
            - redis
            - browserless  # marked

    browserless:  # marked
        image: browserless/chrome  # marked
        restart: always  # marked
        ulimits:  # marked
          core:  # marked
            hard: 0  # marked
            soft: 0  # marked

    redis:
        image: redis:alpine
        restart: always
        volumes:
            - redis-data:/data

volumes:
    redis-data:

```

**Comment by hqwuzhaoyi** [comment]
> How can I add the token to a docker compose file? I currently got
> 
> `JAVDB_SESSION: inserttokenhere`
> 
> but it isnt working

If the session expires, the following content may appear

CleanShot 2024-04-20 at 17 10 54@2x

**Comment by SecondaryH** [comment]
firefox_jMAAhegYhb

I just inserted the token but still getting an error

It does work whenever I replace "censored" by "fc2" though

**Comment by hqwuzhaoyi** [comment]
> firefox_jMAAhegYhb
> 
> I just inserted the token but still getting an error我刚刚插入了令牌但仍然收到错误
> 
> It does work whenever I replace "censored" by "fc2" though每当我用“fc2”替换“censored”时它确实有效

It may also be a network issue. Will trying multiple times have any effect?

**Comment by hqwuzhaoyi** [comment]
> firefox_jMAAhegYhb
> 
> I just inserted the token but still getting an error
> 
> It does work whenever I replace "censored" by "fc2" though

It may also be a network issue. Will trying multiple times have any effect?

**Comment by SecondaryH** [comment]
> > firefox_jMAAhegYhb
> > I just inserted the token but still getting an error
> > It does work whenever I replace "censored" by "fc2" though
> 
> It may also be a network issue. Will trying multiple times have any effect?

I have tried multiple times in the past 24 hours but receive the same error everytime

**Comment by hqwuzhaoyi** [comment]
> > > firefox_jMAAhegYhb
> > > I just inserted the token but still getting an error
> > > It does work whenever I replace "censored" by "fc2" though
> > 
> > 
> > It may also be a network issue. Will trying multiple times have any effect?
> 
> I have tried multiple times in the past 24 hours but receive the same error everytime

Can you clone the project and check what is printed in the 'res' inside this function?

It may be that the token has expired or access to the website has been denied.

lib/routes/javdb/utils.ts

image
